tests/js/client/shell/shell-admin-cluster-rebalance.js

```diff
@@ -144,5 +175,166 @@ function clusterRebalanceSuite() {
   };
 }
 
+function clusterRebalanceOtherOptionsSuite() {
+  let prevDB = null;
+  let dbServer = null;
+  const numCols = 3;
+
+  return {
+
+    setUpAll: function () {
+      prevDB = db._name();
+      db._createDatabase(database);
+      db._useDatabase(database);
+      for (let i = 0; i < numCols; ++i) {
+        db._create("col" + i, {numberOfShards: 2, replicationFactor: 2});
+      }
+
+      let docs = [];
+      for (let i = 0; i < numCols; ++i) {
+        for (let j = 0; j < 1000; ++j) {
+          docs.push({"value": j});
+          if (docs.length > 100) {
+            db["col" + i].insert(docs);
+            docs = [];
+          }
+        }
+        db["col" + i].insert(docs);
+        docs = [];
+      }
+    },
+
+    tearDownAll: function () {
+      db._useDatabase(prevDB);
+      db._dropDatabase(database);
+    },
+
+    testCalcRebalanceStopServer: function () {
+      let dbServers = instanceManager.arangods.filter(arangod => arangod.instanceRole === "dbserver");
+      assertNotEqual(dbServers.length, 0);
+      for (let i = 0; i < dbServers.length; ++i) {
+        dbServer = dbServers[i];
+        assertTrue(suspendExternal(dbServer.pid));
+        let serverHealth = null;
```

Comment:
If the `suspendExternal` is successfull, the respective DB server is suspended. Fine.
However, if one of the test assertions fail, the DB server may stay suspended for the rest of this test file, and potentially also for all following tests in the `shell_client` suite.
So, if this test suspends a DB server, it must make sure that there are no side effects even if the test fails.
So why not wrap everything that follows the `suspendExternal(...)` into a `try { ... } finally { continueExternal(...); wait until server is GOOD again; }`?